recipe/meta.yaml

```diff
@@ -1,311 +1,309 @@
-{% set version = "4.6.0" %}
-{% set build_num = "5" %}
+# This recipe creates 3 packages
+# libopencv, py-opencv, opencv
+#
+# libopencv actually contains all the files, including the site-packages
+# file needed to run opencv with python.
+#     opencv-dnn seems to change on every patch version, therefore, we pin
+#     to the patch version x.x.x
+#
+# py-opencv is a meta package that constrains the python dependencies
+#     This one is pinned to the minor version. x.x
+#
+# opencv is a backward compatible package, with no pinnings
+#
+# This approach is different than the one in defaults. Default's package
+# seperates well the files generated with python dependencies, from those
+# generated without. Unfortunately, this isn't entirely supported by
+# upstream. Their hacks are quite elaborate, making it seem difficult to
+# maintain as opencv has been changing their build system recently.
+#
+# By putting all the generated files in 1 package, this makes the build process
+# much easier, at the expense of a few MBs in the 'lib' package.
+{% set version = "4.10.0" %}
+{% set major_version = version.split('.')[0] %}
+{% set PY_VER_MAJOR = PY_VER.split('.')[0] %}
+{% set PY_VER_MINOR = PY_VER.split('.')[1] %}
+
+{% set build = 0 %}
 
-{% if target_platform != "linux-aarch64" and target_platform != "linux-64"%}
-{%   set enable_testingui = true %}
-{% else %}
-{%   set enable_testingui = false %}
-{% endif %}
 
-{% if build_variant == "headless" %}
-{%   set package_name = "opencv-python-headless" %}
```

Comment:
The recipe use to create a package opencv and another one opencv-python-headless for sf. 

The previous format of the recipe was also to have one real package and two virtual. The one with the binaries used to be opencv, with libopencv and py-opencv being virtual. When headless variant was used, the opencv package was renamed to opencv-python-headless. 

Your current work could be adapted to regain that format. With the global and main output being renamed to opencv (or opencv-python-headless depending on the variant). 

Or we could see how to rebuild packages currently depending on opencv-python-headless to fit this new format.

Reply:
I tried but got this error https://github.com/AnacondaRecipes/opencv-feedstock/pull/31#discussion_r1662223113

Reply:
Rerunning the build after my fix